ldm/models/diffusion/shared_invokeai_diffusion.py

```diff
@@ -0,0 +1,176 @@
+from math import ceil
+from typing import Callable, Optional
+
+import torch
+
+from ldm.models.diffusion.cross_attention_control import CrossAttentionControl
+
+
+class InvokeAIDiffuserComponent:
+    '''
+    The aim of this component is to provide a single place for code that can be applied identically to
+    all InvokeAI diffusion procedures.
+
+    At the moment it includes the following features:
+    * Cross Attention Control ("prompt2prompt")
+    '''
+
+
+    class ExtraConditioningInfo:
```

Comment:
@lstein here's my "ExtraConditioningInfo" object, so that i didn't have to mess with the `c` that gets passed to all those `forward()` functions

Reply:
That's great. Can you show me how to use it?